### PR TITLE
xml-encryption: create typings

### PR DESCRIPTION
**Pull request by midgleyc**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/xml-encryption/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+// Project: https://github.com/auth0/node-xml-encryption#readme
+// Definitions by: Chris Midgley <https://github.com/midgleyc>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+/// <reference types="node" />
+
+export interface EncryptOptions {
+    rsa_pub: string | Buffer;
+    pem: string | Buffer;
+    keyEncryptionAlgorithm: string;
+    encryptionAlgorithm?: string;
```
Comment (peterblazejewicz) [inline types/xml-encryption/index.d.ts:11]: this is closed list, can this be converted to aliases?:
```ts
encryptionAlgorithm?: EncryptionAlgorithm;
```
```ts
type EncryptionAlgorithm = 'http://www.w3.org/2001/04/xmlenc#aes128-cbc' | '...' | ....;
```
https://github.com/auth0/node-xml-encryption/blob/master/lib/xmlenc.js#L99

**Review by peterblazejewicz — COMMENTED** [review]
the `callback` function/method is more like Node exemplary callback, so it
should accept null as another optin for error:
https://github.com/auth0/node-xml-encryption/blob/master/lib/xmlenc.js#L108
so someting like (pseudo):
```ts
callback?: (error: Error | null, results: string) => void
```

File: types/xml-encryption/index.d.ts (hunk 1 above)
Comment (midgleyc) [reply types/xml-encryption/index.d.ts]: yes, good spot, thanks

**Comment by orta** [comment]
Cool, looks reasonable to me 👍🏾 - thanks @midgleyc

**Pull request merged**